app/src/main/java/org/oppia/android/app/profile/ProfileLoginFragmentPresenter.kt

```diff
@@ -421,8 +423,17 @@ class ProfileLoginFragmentPresenter @Inject constructor(
       .showNow(activity.supportFragmentManager, TAG_VALIDATE_ADMIN_PIN_DIALOG)
   }
 
+  /**
+   * Deletes all app data by removing all profile,s resetting onboarding state, and closing the app.
+   *
+   * Once completed, the user will be forced to go through onboarding again to re-create their
+   * profile.
+   */
   private fun deleteAppData() {
     profileManagementController.deleteAllProfiles().toLiveData().observe(fragment) {
+      // Reset onboarding so the app starts fresh.
+      appStartupStateController.resetOnboardingState()
```

Comment:
Perhaps worth verifying via test that this step happens?

Reply:
Added, and here is proof that the test fails if the wrong `StartupMode` is asserted.

<img width="1326" height="504" alt="Screenshot 2025-10-02 at 22 11 10" src="https://github.com/user-attachments/assets/b72c301a-86b3-4c3d-9858-9a6dea20fb10" />